### PR TITLE
Reposync speedup part 3

### PR DESCRIPTION
**Pull request by moio**
## What does this PR change?

It replaces the existing mechanism in `spacewalk-repo-sync` to insert "non-special" per-package table rows (anything other than capabilities, changelogs and checksums) with a more efficient one.

**Total speedup is ~140%**, individual contributions from relevant commits as measured in my setup (syncing `sles11-sp4-updates-x86_64` from a local mirror, robustly transferable to other channels) follow:

| Commit                                                             | Speedup |
|--------------------------------------------------------------------|---------|
| reposync speedup: use execute_values for INSERTS of generic tables | 77.58%  |
| reposync speedup: use fast execution helper in executemany         | 64.02%  |

## Change details

Insert batches before this patch used the `executemany` method from `psycopg2`, which is notoriously slow, and this PR refactors code to use the faster `execute_values` and `execute_batch` instead. Per the library documentation:

> The current implementation of executemany() is (using an extremely charitable understatement) not particularly performing. [The alternative execute_values and execute_batch methods] can be used to speed up the repeated execution of a statement against a set of parameters. By reducing the number of server roundtrips the performance can be orders of magnitude better than using executemany()."

Source: https://www.psycopg.org/docs/extras.html#fast-execution-helpers

## Dependencies

This change requires the latest psycopg2 library in order to work, and builds on top of #2157 for this reason.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **internal change only**
- [x] **DONE**

## Test coverage
- No tests: **Cucumber tests to be implemented after reposync is fast enough**

- [x] **DONE**

## Links

Fixes partially https://github.com/SUSE/spacewalk/issues/9275
- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by moio** [comment]
@mcalmer the only clear downside of this patch that I can see is that the newer methods do not return an accurate count of the affected rows. We did not really use this number in code being replaced apart from some debugging statements, which I have removed, so the deal shouldn't be so big - but there is no easy way around that.

**Comment by moio** [comment]
Best reviewed commit-by-commit as others in this series.

**Comment by mcalmer** [comment]
@moio already forgot in "Part 2": If it requires a special version of psycopg2 we should make a dependency in the specfile using the first upstream version which had it.

File: backend/server/importlib/backendLib.py [hunk 1]
```diff
@@ -432,31 +432,27 @@ class TableInsert(TableUpdate):
...
         else:
-            chunksize = self.count
             blob_map = None
 
         # Do the insert
         statement = self._getCachedQuery(None, blob_map=blob_map)
-        executeStatement(statement, values, chunksize)
+        l = len(values[self.insert_fields[0]])
+        value_list = [[values[f][i] for f in self.insert_fields] for i in range(l)]
+        statement.execute_values(self._buildQuery(None), value_list, fetch=False, page_size=10_000)
```
Comment (mcalmer) [inline backend/server/importlib/backendLib.py:453]: Is the `_` underscore in the page_type value a typo?

Comment (moio) [reply backend/server/importlib/backendLib.py:453]: No, it's the actual parameter name in psycopg2, specifically:

https://github.com/psycopg/psycopg2/commit/9a8d7d5a6ad5f24e1cbc529e98dd61ad9424b9ed#diff-e6256dfa32f9bced2faae91403f4d594R1201

**Comment by moio** [comment]
> @moio already forgot in "Part 2": If it requires a special version of psycopg2 we should make a dependency in the specfile using the first upstream version which had it.

Right, I fixed it now.

File: backend/server/importlib/backendLib.py (hunk 1 above)
Comment (mcalmer) [reply backend/server/importlib/backendLib.py:453]: oh :-)

**Comment by moio** [comment]
Thanks for the speedy review! All tests passed so I am merging. Most interesting (and final!) part 4 is coming before evening as well :wink:

**Pull request merged**